OMEdit/OMEditGUI/Git/GitCommands.cpp

```diff
@@ -0,0 +1,294 @@
+  mpGitProcess->start("git", QStringList() << "status"<< "--porcelain" << filePath);
+  mpGitProcess->waitForFinished();
+  QByteArray changedFilesOutput =  mpGitProcess->readAllStandardOutput();
+  QStringList changedFilesOutputList = QString(changedFilesOutput).split("\n");
+  return changedFilesOutputList;
+}
+
+/*!
+ * \brief GitCommands::getRepositoryName
+ * Returns the repository .
+ * \return
+ */
+QString GitCommands::getRepositoryName(QString filePath)
+{
+  QFileInfo fileInfo(filePath);
+  QString repository = fileInfo.absoluteDir().absolutePath();
+  mpGitProcess->setWorkingDirectory(repository);
+  mpGitProcess->start("git", QStringList() << "rev-parse" << "--show-toplevel");
+  mpGitProcess->waitForFinished();
+  QString repositoryName =  mpGitProcess->readAllStandardOutput();
+  return repositoryName;
+}
+
+/*!
+ * \brief GitCommands::getBranchName
+ * Returns the branch name .
+ * \return
+ */
+QString GitCommands::getBranchName(QString filePath)
+{
+  QFileInfo fileInfo(filePath);
+  QString repository = fileInfo.absoluteDir().absolutePath();
+  mpGitProcess->setWorkingDirectory(repository);
+  mpGitProcess->start("git", QStringList() << "rev-parse" << "--abbrev-ref" << "HEAD");
+  mpGitProcess->waitForFinished();
+  QString branchName =  mpGitProcess->readAllStandardOutput();
+  return branchName;
+}
+
+/*!
+ * \brief GitCommands::getAuthorName
+ * Returns the author name .
+ * \return
+ */
+QString GitCommands::getAuthorName()
+{
+  mpGitProcess->start("git", QStringList() << "config" << "user.name");
+  mpGitProcess->waitForFinished();
+  QString author =  mpGitProcess->readAllStandardOutput();
+  return author;
+}
+
+/*!
+ * \brief GitCommands::getEmailName
+ * Returns the email name .
+ * \return
+ */
+QString GitCommands::getEmailName()
+{
+  mpGitProcess->start("git", QStringList() << "config" << "user.email");
+  mpGitProcess->waitForFinished();
+  QString email =  mpGitProcess->readAllStandardOutput();
+  return email;
+}
+
+/*!
+ * \brief GitCommands::getGitHash
+ * Returns the git hash of the file .
+ * \return
+ */
+QString GitCommands::getGitHash(QString fileName)
+{
+  QFileInfo fileInfo(fileName);
+  QString filePath = fileInfo.absoluteDir().absolutePath();
+  mpGitProcess->setWorkingDirectory(filePath);
+  mpGitProcess->start("git", QStringList() << "hash-object" << fileName);
+  mpGitProcess->waitForFinished();
+  QString gitHash =  mpGitProcess->readAllStandardOutput();
+  return gitHash;
+}
+
+/*!
+ * \brief GitCommands::commitFiles
+ * commit files to the local repository.
+ * \param repositoryPath
+ */
+void GitCommands::commitFiles(QString repositoryPath, QString commitMessage)
+{
+  QFileInfo fileInfo(repositoryPath);
+  QString directory = fileInfo.absoluteDir().absolutePath();
+  mpGitProcess->setWorkingDirectory(directory);
+  mpGitProcess->start("git", QStringList() << "commit" <<"-m" << commitMessage);
+  mpGitProcess->waitForFinished();
+}
```

Comment:
The git interface is not done nicely. There is no way to check if git command has failed. You are always assuming that the commands are successful. You need to add some logger for git commands and outputs so user knows what's going on. This will also help you. You also need to check if git command has failed and return true or false based on it. You don't want to continue doing GUI operations if e.g., `git add` command has failed.